custom/icds_reports/utils/aggregation_helpers/distributed/agg_gov_dashboard.py

```diff
@@ -0,0 +1,194 @@
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER<=36 THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS open_all_0_3_female,
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER<=36 THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS open_all_0_3_male,
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
+                    valid_in_month ELSE 0 END ) AS valid_all_3_6_female,
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
+                    valid_in_month ELSE 0 END ) AS valid_all_3_6_male,
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS open_all_3_6_female,
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS open_all_3_6_male,
+
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER<=36 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_0_3_female,
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER<=36 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_0_3_male,
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER<=36 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS open_reg_in_month_0_3_female,
+
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER<=36 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS open_reg_in_month_0_3_male,
+
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_female,
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_male,
+
+                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_female,
+
+                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 AND
+                    date_trunc('month',opened_on)=%(start_date)s THEN
+                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_male
+            FROM child_health_monthly
```

Comment:
Used child_health_monthly because Cost is lesser than hitting agg_child health on agg_level=5

Reply:
Out of curiosity, what would the query look like if it was coming from agg_child_health?

Reply:
Sure:. Below is the comparison of similar query on child_health_monthly and agg_child_health( I just removed `opened_on where clause` as this column is new. Query should give good idea anyway.

ON `child_health_monthly`
```
SELECT
                supervisor_id,
                awc_id,
                sum(CASE WHEN sex='F' AND age_tranche::INTEGER<=36 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_0_3_female,
                sum(CASE WHEN sex='M' AND age_tranche::INTEGER<=36 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_0_3_male,
                sum(CASE WHEN sex='F' AND age_tranche::INTEGER<=36 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_0_3_female,
                sum(CASE WHEN sex='M' AND age_tranche::INTEGER<=36 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_0_3_male,
                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_3_6_female,
                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_3_6_male,
                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_3_6_female,
                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_3_6_male,

                sum(CASE WHEN sex='F' AND age_tranche::INTEGER<=36  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_0_3_female,
                sum(CASE WHEN sex='M' AND age_tranche::INTEGER<=36  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_0_3_male,
                sum(CASE WHEN sex='F' AND age_tranche::INTEGER<=36  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_reg_in_month_0_3_female,

                sum(CASE WHEN sex='M' AND age_tranche::INTEGER<=36  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_reg_in_month_0_3_male,

                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_female,
                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_male,

                sum(CASE WHEN sex='F' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_female,

                sum(CASE WHEN sex='M' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_male
            FROM child_health_monthly
            WHERE month='2020-01-01'
            GROUP BY supervisor_id, awc_id;

/*
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 HashAggregate  (cost=0.00..0.00 rows=0 width=0)
   Group Key: remote_scan.supervisor_id, remote_scan.awc_id
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=6432 dbname=icds_ucr
               ->  Finalize GroupAggregate  (cost=580760.86..747338.24 rows=561822 width=194)
                     Group Key: supervisor_id, awc_id
                     ->  Gather Merge  (cost=580760.86..704561.27 rows=825750 width=130)
                           Workers Planned: 5
                           ->  Partial GroupAggregate  (cost=579760.78..604120.41 rows=165150 width=130)
                                 Group Key: supervisor_id, awc_id
                                 ->  Sort  (cost=579760.78..580173.66 rows=165150 width=78)
                                       Sort Key: supervisor_id, awc_id
                                       ->  Parallel Index Scan using chm_month_supervisor_id_102648 on child_health_monthly_102648 child_health_monthly  (cost=0.56..556850.01 rows=165150 width=78)
                                             Index Cond: (month = '2020-01-01'::date)
(17 rows)

*/

```


On agg_child_health

```

SELECT
                supervisor_id,
                awc_id,
                sum(CASE WHEN gender='F' AND age_tranche::INTEGER<=36 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_0_3_female,
                sum(CASE WHEN gender='M' AND age_tranche::INTEGER<=36 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_0_3_male,
                sum(CASE WHEN gender='F' AND age_tranche::INTEGER<=36 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_0_3_female,
                sum(CASE WHEN gender='M' AND age_tranche::INTEGER<=36 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_0_3_male,
                sum(CASE WHEN gender='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_3_6_female,
                sum(CASE WHEN gender='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_in_month ELSE 0 END ) AS valid_all_3_6_male,
                sum(CASE WHEN gender='F' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_3_6_female,
                sum(CASE WHEN gender='M' AND age_tranche::INTEGER BETWEEN 37 AND 72 THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_all_3_6_male,

                sum(CASE WHEN gender='F' AND age_tranche::INTEGER<=36  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_0_3_female,
                sum(CASE WHEN gender='M' AND age_tranche::INTEGER<=36  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_0_3_male,
                sum(CASE WHEN gender='F' AND age_tranche::INTEGER<=36  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_reg_in_month_0_3_female,

                sum(CASE WHEN gender='M' AND age_tranche::INTEGER<=36  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS open_reg_in_month_0_3_male,

                sum(CASE WHEN gender='F' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_female,
                sum(CASE WHEN gender='M' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_in_month ELSE 0 END ) AS valid_reg_in_month_3_6_male,

                sum(CASE WHEN gender='F' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_female,

                sum(CASE WHEN gender='M' AND age_tranche::INTEGER BETWEEN 37 AND 72  THEN
                    valid_all_registered_in_month ELSE 0 END ) AS  open_reg_in_month_3_6_male
            FROM agg_child_health
            WHERE month='2020-01-01' and aggregation_level=5
            GROUP BY supervisor_id, awc_id;

/*
 Finalize GroupAggregate  (cost=1975173.43..2603443.26 rows=40000 width=194)
   Group Key: "agg_child_health_2020-01-01".supervisor_id, "agg_child_health_2020-01-01".awc_id
   ->  Gather Merge  (cost=1975173.43..2595843.26 rows=160000 width=130)
         Workers Planned: 4
         ->  Partial GroupAggregate  (cost=1974173.38..2575785.66 rows=40000 width=130)
               Group Key: "agg_child_health_2020-01-01".supervisor_id, "agg_child_health_2020-01-01".awc_id
               ->  Sort  (cost=1974173.38..1985104.51 rows=4372453 width=78)
                     Sort Key: "agg_child_health_2020-01-01".supervisor_id, "agg_child_health_2020-01-01".awc_id
                     ->  Parallel Append  (cost=0.00..1150505.06 rows=4372453 width=78)
                           ->  Parallel Seq Scan on "agg_child_health_2020-01-01"  (cost=0.00..1128642.79 rows=4372453 width=78)
                                 Filter: ((month = '2020-01-01'::date) AND (aggregation_level = 5))
                           ->  Parallel Seq Scan on agg_child_health  (cost=0.00..0.00 rows=1 width=136)
                                 Filter: ((month = '2020-01-01'::date) AND (aggregation_level = 5))
(13 rows)

*/

```